src/NuGet.Clients/NuGet.SolutionRestoreManager/VsSolutionRestoreService.cs

```diff
@@ -387,6 +390,23 @@ private static VersionRange GetVersionRange(IVsReferenceItem item)
             return VersionRange.All;
         }
 
+        private static bool GetPropertyBoolOrFalse(
+                IVsReferenceItem item, string propertyName)
+        {
+            try
+            {
+                return MSBuildStringUtility.IsTrue(item.Properties?.Item(propertyName)?.Value);
+            }
```

Comment:
Could it also throw InvalidCastException as we saw in case of TargetFramework/ TargetFrameworks properties value?